### DISCUSSION TITLE
Support for DCAT 3

### DISCUSSION THREAD

**Discussion by jze** [ckan/ckan#7855] [General]
[DCAT 3](https://www.w3.org/TR/vocab-dcat-3/) (and the European draft [DCAT-AP 3](https://semiceu.github.io/DCAT-AP/releases/3.0.0/)) changes the structure of metadata significantly. Instead of *dataset* → *distribution* (or *package* → *resource* in CKAN terms) we now have *datasets*, *dataset series* and *data services* as first-class citizen entities. We are thus getting closer to the geo-world and ISO metadata. I wonder if we can even model this with the current structures in CKAN.

*Data services* have already been introduces in DCAT 2. However, it seems to me that they are not yet first-class citizen entities with an own entry in the `package` table. Instead they can just be found as JSON data in the `package_extra` table entry of a dataset.

The `package` table already contains a column `type` that might be used to distinguish between *dataset*, *dataset series* and *data service*. CKAN even uses the term `dataset` at this point today. The column is also used to indicate harvest sources (that are stored in the `package` table, too). I suspect, however, that major modifications to CKAN itself will be necessary to display these different entities. 

I have also created an [issue for the DCAT extension](https://github.com/ckan/ckanext-dcat/issues/257) but this place is maybe more appropriate.

**Comment by inderps** [comment]
I strongly back this proposal. Prioritizing support for first-class entities—datasets, dataset series, data services—as per DCAT 3 is crucial for CKAN's relevance. This shift is not just an enhancement; it's essential to meet the evolving needs of users and align with standard metadata practices.

It's a deal-breaker for many catalogues, and addressing it shouldn't be delayed.

**Comment by jqnatividad** [comment]
FYI, DCAT-US is also being updated to align with DCAT 3. 

https://github.com/GSA/resources.data.gov/issues/621

IMHO, this should further lend weight to prioritizing synching our efforts with both initiatives so CKAN will be DCAT 3 ready by the time both the DCAT-AP v3 and DCAT-US v3 specifications are published.

**Comment by wardi** [comment]
Has anyone tried implementing dataset, dataset series and data service as separate package types using ckanext-scheming? Are there any features missing for fully implementing series and data services records?

**Comment by jqnatividad** [comment]
IMHO, this is a good topic to convene a Working Group/Community of Practice as envisioned for Phase II of the NSF CKAN POSE project (https://civicdataecosystem.org/).

And I don't think we need to wait for POSE Phase II to be approved given the importance of the topic.

Speaking of which, [DCAT-US v3](https://github.com/DOI-DO/dcat-us) has an open comment period till Nov 17.

This might be a good time to put in your 2 cents and to take a pulse of what data publishers need.

Given that the spec aims to "_provide a **single metadata standard** able to support most requirements for documentation of business, technical, statistical, and geospatial data consistently_", it will greatly expand CKAN's utility and addressable use cases.

cc @thegostev @anuveyatsu @gavram @rgradeck @drw @saylorsd @rossreillypitt @samibaig

**Comment by Zharktas** [comment]
Just to note here as well, we've implemented datasets and data services in our DCAT profile with ckan extensions using scheming, the extension is available here https://github.com/vrk-kpa/ckanext-apis which then is parsed in the profile as data services. For example https://www.avoindata.fi/data/fi/dataset/50bf822d-31d4-450c-917d-c974af92fab6.xml

**Reply by AlasdairGray** [reply]
Thanks for sharing, there seems to be some semantic errors in your RDF. You seem to be mixing properties for describing a `dcat:Dataset` and `dcatDistribution`. For example, the property `dcat:distribution` is declared for the `dcat:Dataset` class rather than the `dcat:DataService` class. Also the `dcat:endpointURL` and `dcat:downloadURL` should point to different resources, the latter being the direct link to the downloadable resource rather than an API through which you retrieve data.

**Comment by jqnatividad** [comment]
Following up on the DCAT 3 discussion during the Tech Team call this morning - here's a wonderful overview of the DCAT 3 Standard by @fellahst.

https://medium.com/@stephanef/unraveling-the-dcat-standard-and-the-benefits-of-a-dcat-us-profile-c2c5b6cfbd44

Though it is through the DCAT-US 3 lens, its gives the "Big Picture" and covers the DCAT-AP/GeoDCAT-AP initiatives in Europe as well.

**Comment by anuveyatsu** [comment]
This sounds related to https://github.com/ckan/ckan/issues/7489 where we started discussion based on DCAT-AP v2. This comment from @amercader already indicates some constraints with `DataService` - https://github.com/ckan/ckan/issues/7489#issuecomment-1525430371

Based on above discussion and ideas about the approach, I believe it would be good to consider this under [CKAN 3.0 Taskforce](https://github.com/orgs/ckan/projects/3/views/1). cc @thegostev @gavram

**Comment by thegostev** [comment]
@anuveyatsu thanks for the suggestion, it is a great addition to the scope of CKAN 3.0: https://github.com/ckan/ckan/issues/7896

I do agree that it's a high priority.

**Comment by jqnatividad** [comment]
Great! DCAT 3.0 + CKAN 3.0 would make for a compelling upgrade and present an opportunity to revisit/modernize architectural choices.

Taking  a cue from the [DCAT-US 3.0 abstract](https://doi-do.github.io/dcat-us/#abstract) note:

> ### NOTE
> DCAT 3 advances beyond its predecessor, DCAT 2 Data Catalog Vocabulary (DCAT) - Version 2, while preserving its relevance without rendering it obsolete. The foundational DCAT namespace persists in DCAT 3, ensuring a harmonious coexistence with the terminology of DCAT 2. Notably, while DCAT 3 introduces enhancements such as relaxed constraints, novel classes, and properties, these modifications uphold the integrity of prior terms, preventing any disruption to their definitions.
> 
> In the realm of implementations, any **_fresh deployment is expected to embrace the DCAT 3 standard_**, harnessing its evolved capabilities. However, pre-existing implementations need not undergo an upgrade unless they seek to leverage the newfound functionalities. **_Specifically, current instances of DCAT 2/ POD 1.1 deployments that do not intersect with DCAT 3's novel elements—such as versioning, dataset series, and inverse properties—can continue without alteration, maintaining full compliance with the DCAT 3 specifications._** (highlights mine)

Perhaps, CKAN 3 can introduce **first class support** for key DCAT 3 novel elements - i.e. versioning, dataset series, inverse properties, geospatial/temporal data, etc. that *may* require breaking changes?

That is, instead of depending on the ckanext-dcat  **_extension_**, bring these DCAT 3 novel elements into CKAN 3.0 core?

So long as we have a CKAN 2.x to 3.0 migration path and  "DCAT 3.0-light" is available on CKAN 2.x by building on the existing ckanext-dcat extension...

Doing so, IMHO, will give architectural impetus to CKAN 3.0 given DCAT-US 3.0's **_future-proof_** ambitions.

Again, per the last paragraph of the DCAT-US 3.0 abstract:

> DCAT-US 3.0 introduces a highly refined, interoperable, and **_future-proof framework_** for describing and validating dataset metadata. In essence, it is **_not just a specification but an advanced stride towards achieving a data-centric landscape_** where precise metadata description empowers the efficient flow of information while **_laying the groundwork for sustained innovation_**. (highlights mine)

Of course, I realize these quotes are from the DCAT-US 3.0 draft spec which is but one profile. But what if by aligning CKAN 3.0 to DCAT 3.0, CKAN 3.0 can load any DCAT 3.0 profile?

https://www.w3.org/TR/vocab-dcat-3/#profiles
https://www.w3.org/TR/vocab-dcat-3/#conformance

IMHO, DCAT 3.0's alignment with FAIR initiatives also help further widen CKAN's addressable market beyond traditional government data portals - specifically, scientific/research data portals and enterprise data catalogs - possibly, with their own corresponding, industry/domain/organization-specific DCAT 3.0 profiles.

Just my 2 cents 😉

**Comment by thegostev** [comment]
The recent update of ckanext-DCAT enabled CKAN to comply with [DCAT-AP 3.0](https://semiceu.github.io/DCAT-AP/releases/3.0.0/) and DCAT-US v.3. 

Looking forward to your feedback!
